invokeai/app/invocations/generate.py

```diff
@@ -57,19 +58,24 @@ def step_callback(sample, step=0):
         # Handle invalid model parameter
         # TODO: figure out if this can be done via a validator that uses the model_cache
         # TODO: How to get the default model name now?
-        if self.model is None or self.model == "":
-            self.model = context.services.generate.model_name
+        factory = context.services.generator_factory
+        if self.model:
+            factory.model_name = self.model
+        else:
+            self.model = factory.model_name
```

Comment:
I agree! I just adapted what @Kyle0654 had there originally. There's a design decision to be made here about where the default model is coming from. The legacy logic is:

1) if the model is specified on the command line's `--model` argument, then use that.
2) otherwise let the model manager select the default model, by first picking the `default: True` stanza, or if there is none, then taking the first stanza that appears in `models.yaml`.

It really depends on how the starting model is selected by the nodes clients, and I punted here.